docker/docker_client.go

```diff
@@ -243,7 +243,13 @@ func (c *dockerClient) getBearerToken(realm, service, scope string) (string, err
 	return tokenStruct.Token, nil
 }
 
-func getAuth(hostname string) (string, string, error) {
+func getAuth(ctx *types.SystemContext, hostname string) (string, string, error) {
+	if ctx.DockerAuthConfig != nil {
+		if ctx.DockerAuthConfig.Auth != "" {
+			return decodeDockerAuth(ctx.DockerAuthConfig.Auth)
+		}
+		return ctx.DockerAuthConfig.Username, ctx.DockerAuthConfig.Password, nil
```

Comment:
Ocid can provide username/password from k8s or skopeo can. K8s and OpenShift can just provide Auth.